jolt-core/src/zkvm/spartan/shift.rs

```diff
@@ -18,80 +22,56 @@ use crate::transcripts::Transcript;
 use crate::utils::math::Math;
 use crate::zkvm::bytecode::BytecodePreprocessing;
 use crate::zkvm::dag::state_manager::StateManager;
-use crate::zkvm::instruction::{CircuitFlags, InstructionFlags};
-use crate::zkvm::r1cs::inputs::{
-    evaluate_shift_sumcheck_witnesses, generate_shift_sumcheck_witnesses,
-};
+use crate::zkvm::instruction::{CircuitFlags, Flags, InstructionFlags};
 use crate::zkvm::r1cs::key::UniformSpartanKey;
 use crate::zkvm::witness::VirtualPolynomial;
 use rayon::prelude::*;
 
-#[derive(Allocative)]
-struct ShiftSumcheckProverState<F: JoltField> {
-    combined_witness_poly: MultilinearPolynomial<F>,
-    is_noop_poly: MultilinearPolynomial<F>,
-    eq_plus_one_r_cycle: MultilinearPolynomial<F>,
-    eq_plus_one_r_product: MultilinearPolynomial<F>,
-    #[allocative(skip)]
-    trace: Arc<Vec<Cycle>>,
-    #[allocative(skip)]
-    bytecode_preprocessing: BytecodePreprocessing,
-}
+/// Sumcheck round poly degree.
+const DEGREE: usize = 2;
 
 #[derive(Allocative)]
 pub struct ShiftSumcheck<F: JoltField> {
     gamma_powers: [F; 5],
     log_T: usize,
-    prover_state: Option<ShiftSumcheckProverState<F>>,
+    prover: Option<Prover<F>>,
```

Comment:
nit: we have `prover_state` everywhere in the codebase. 

```suggestion
    prover_state: Option<ShifgtSumcheckProverState<F>>,
```